editor/effects.js

```diff
@@ -217,6 +218,9 @@ export default {
 		// Only focus the previous block if it's not mergeable
 		if ( ! blockType.merge ) {
 			dispatch( focusBlock( blockA.uid ) );
+			if ( blockB.name === 'core/paragraph' && ! blockB.attributes.content.length ) {
```

Comment:
My instinct here is to invent a `isBlockEmpty` helper which we would use instead of this very very specific logic. 

I'm not sure, though, since it probably _is_ only paragraphs that we want this behaviour for...

Thoughts, @aduth?

Reply:
We could also introduce a flag that marks all removable blocks like heading or list. 

`blockB.attributes.content.length` - is it safe to assume that content is there and it's a string?

Reply:
And yes, you can at least introduce a local variable which better explains what is happening in this conditional statement :)

Reply:
Maybe we could introduce a ìsEmpty` property to the block API and if not defined the block is never empty. Only the block knows how to read its attributes.

Reply:
I added an `isEmpty` property and implemented it for `core/paragraph`!

Reply:
I don't love the `isEmpty` property (more to a general concern of limiting number of properties on the block API). It seems to me if Editable is smart enough to know that it should merge its own content into previous, it can also be smart enough to know that it has no content to merge and should just remove. That said, I also don't love an `onRemove` being added to Editable. Wondering if we could enhance one of the existing like `onReplace` to also encompass behavior of "replacing with nothing" ?

Reply:
That's a good idea @aduth. I moved my deletion logic to `Editable` in a0d193b7291da64bc9ce795f0f185e979b1bcbf9.